src/js/tabs/failsafe.js

```diff
@@ -297,9 +297,18 @@ failsafe.initialize = function (callback) {
 
         if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
             if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_46)) {
-                $('input[name="gps_rescue_min_dth"]').attr("min", 10);
-            } else if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_45)) {
-                $('input[name="gps_rescue_min_dth"]').attr("min", 20);
+                $('input[name="gps_rescue_min_dth"]').attr({"min": 10, "max": 30});
+                $('input[name="gps_rescue_ground_speed"]').attr({"min": 3.0, "max": 30.0, "step": 0.1});
+                $('input[name="gps_rescue_ascend_rate"]').attr({"min": 1.0, "max": 25.0, "step": 0.1});
+                $('input[name="gps_rescue_descend_rate"]').attr({"min": 1.0, "max": 5.0, "step": 0.1});
+                $('input[name="gps_rescue_descent_distance"]').attr("min", 10);
+                $('input[name="gps_rescue_initial_climb"]').val(FC.GPS_RESCUE.initialClimbM);
+            } else {
+                $('input[name="gps_rescue_initial_climb"]').closest('.number').hide();
+
+                if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_45)) {
+                    $('input[name="gps_rescue_min_dth"]').attr("min", 20);
+                }
             }
```

Comment:
Please refactor this code in a simpler but functionally equivalent form, so that we check each version sequentially.
This makes it much easier to review and manage the settings.
The following code provides correct values for 4.4 and 4.5, and I strongly recommend using it:
```
        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
            $('input[name="gps_rescue_min_dth"]').attr("min", 50);
            $('input[name="gps_rescue_min_dth"]').val(FC.GPS_RESCUE.minRescueDth);
        } else {
            $('input[name="gps_rescue_min_dth"]').closest('.number').hide();
        }

        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_45)) {
            $('input[name="gps_rescue_min_dth"]').attr("min", 20);
            $('input[name="gps_rescue_ascend_rate"]').attr("min", 0.5);
            $('input[name="gps_rescue_ascend_rate"]').attr("max", 25);
            $('input[name="gps_rescue_descent_distance"]').attr("min", 5);
            $('input[name="gps_rescue_descend_rate"]').attr("min", 0.25);
            $('input[name="gps_rescue_return_altitude"]').attr("min", 2);
            $('input[name="gps_rescue_return_altitude"]').attr("max", 255);
            $('input[name="gps_rescue_ground_speed"]').attr("min", 0);
            $('input[name="gps_rescue_ground_speed"]').attr("max", 30);
            $('input[name="gps_rescue_angle"]').attr("max", 80);
        }

        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_46)) {
            $('input[name="gps_rescue_min_dth"]').attr("min", 10);
            $('input[name="gps_rescue_min_dth"]').attr("max", 30);
            $('input[name="gps_rescue_return_altitude"]').attr("min", 5);
            $('input[name="gps_rescue_return_altitude"]').attr("max", 1000);
            $('input[name="gps_rescue_angle"]').attr("min", 30);
            $('input[name="gps_rescue_angle"]').attr("max", 60);
            $('input[name="gps_rescue_descent_distance"]').attr("min", 10);
            $('input[name="gps_rescue_initial_climb"]').attr("min", 0);
            $('input[name="gps_rescue_initial_climb"]').attr("max", 100);
            $('input[name="gps_rescue_initial_climb"]').val(FC.GPS_RESCUE.initialClimbM);
        } else {
            $('input[name="gps_rescue_initial_climb"]').closest('.number').hide();
        }
```